src/arrayMethodJoin.js

```diff
@@ -5,7 +5,37 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let separ = separator;
+
+    switch (separ) {
+      case undefined:
+        separ = ',';
+        break;
+
+      case null:
+        separ = 'null';
+        break;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] === undefined) {
+        continue;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        result += ',';
+        continue;
+      }
+
+      if (i === this.length - 1) {
+        result += String(this[i]);
+      } else {
+        result += String(this[i]) + separ;
+      }
```

Comment:
Don't repeat yourself (result += ...) : just add element to the result for all cases and add separator for all elements instead of the last